tests/pytests/functional/loader/test_loader.py

```diff
@@ -26,6 +26,66 @@ def venv(tmp_path):
         yield _venv
 
 
+@pytest.fixture
+def module_dirs(tmp_path):
+    module_dir = tmp_path / "module-dir-base"
+    module_dir.joinpath("modules").mkdir(parents=True)
+    return [str(module_dir)]
+
+
+def test_module_dirs_priority(venv, salt_extension, minion_opts, module_dirs):
+    # Install our extension into the virtualenv
+    venv.install(str(salt_extension.srcdir))
+    installed_packages = venv.get_installed_packages()
+    assert salt_extension.name in installed_packages
+    code = """
+    import sys
+    import json
+    import salt._logging
+    import salt.loader
+
+    minion_config = json.loads(sys.stdin.read())
+    salt._logging.set_logging_options_dict(minion_config)
+    salt._logging.setup_logging()
+    mod_dirs = salt.loader._module_dirs(minion_config, "modules", "module")
+    print(json.dumps(mod_dirs))
+    """
+    minion_opts["module_dirs"] = module_dirs
+    ret = venv.run_code(code, input=json.dumps(minion_opts))
+    module_dirs_return = json.loads(ret.stdout)
+    assert len(module_dirs_return) == 5
+    for i, tail in enumerate(
+        [
+            "/module-dir-base/modules",
+            "/var/cache/salt/minion/extmods/modules",
+            "/module-dir-base",
+            "/site-packages/salt_ext_loader_test/modules",
+            "/site-packages/salt/modules",
+        ]
+    ):
+        assert module_dirs_return[i].endswith(
+            tail
+        ), f"{module_dirs_return[i]} does not end with {tail}"
+
+    # Test the deprecated mode as well
+    minion_opts["features"] = {"enable_deprecated_module_search_path_priority": True}
+    ret = venv.run_code(code, input=json.dumps(minion_opts))
+    module_dirs_return = json.loads(ret.stdout)
+    assert len(module_dirs_return) == 5
+    for i, tail in enumerate(
+        [
+            "/module-dir-base/modules",
+            "/module-dir-base",
+            "/site-packages/salt_ext_loader_test/modules",
+            "/var/cache/salt/minion/extmods/modules",
+            "/site-packages/salt/modules",
+        ]
+    ):
+        assert module_dirs_return[i].endswith(
+            tail
+        ), f"{module_dirs_return[i]} does not end with {tail}"
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "enable_deprecated_module_search_path_priority",
    [False, True],
    ids=lambda x: "enable_deprecated_module_search_path_priority={}".format(x),
)
def test_module_dirs_priority(venv, salt_extension, minion_opts, module_dirs, enable_deprecated_module_search_path_priority):
    # Install our extension into the virtualenv
    venv.install(str(salt_extension.srcdir))
    installed_packages = venv.get_installed_packages()
    assert salt_extension.name in installed_packages
    code = """
    import sys
    import json
    import salt._logging
    import salt.loader

    minion_config = json.loads(sys.stdin.read())
    salt._logging.set_logging_options_dict(minion_config)
    salt._logging.setup_logging()
    mod_dirs = salt.loader._module_dirs(minion_config, "modules", "module")
    print(json.dumps(mod_dirs))
    """
    minion_opts["module_dirs"] = module_dirs
    minion_opts["features"] = {"enable_deprecated_module_search_path_priority": enable_deprecated_module_search_path_priority}
    if enable_deprecated_module_search_path_priority is False:
        expected = [
            "/module-dir-base/modules",
            "/var/cache/salt/minion/extmods/modules",
            "/module-dir-base",
            "/site-packages/salt_ext_loader_test/modules",
            "/site-packages/salt/modules",
        ]
    else:
        expected = [
            "/module-dir-base/modules",
            "/module-dir-base",
            "/site-packages/salt_ext_loader_test/modules",
            "/var/cache/salt/minion/extmods/modules",
            "/site-packages/salt/modules",
        ]
    ret = venv.run_code(code, input=json.dumps(minion_opts))
    module_dirs_return = json.loads(ret.stdout)
    assert len(module_dirs_return) == 5
    for i, tail in enumerate(expected)
        assert module_dirs_return[i].endswith(
            tail
        ), f"{module_dirs_return[i]} does not end with {tail}"
```

Reply:
I can do that if you insist, but I optimized the code for easy deletion, so it will be straightforward to remove the deprecated bits in a future release.